File: app/src/main/java/mb00/android/codehub/ui/user/adapter/UserFragmentPagerAdapter.java
package mb00.android.codehub.ui.user.adapter;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import mb00.android.codehub.ui.user.view.UserActivity;
import mb00.android.codehub.ui.user.view.UserFollowersFragment;
import mb00.android.codehub.ui.user.view.UserFollowingFragment;
import mb00.android.codehub.ui.user.view.UserGistsFragment;
import mb00.android.codehub.ui.user.view.UserOverviewFragment;
import mb00.android.codehub.ui.user.view.UserPulseFragment;
import mb00.android.codehub.ui.user.view.UserReposFragment;
import mb00.android.codehub.ui.user.view.UserStarredFragment;

/**
 * FragmentPagerAdapter used to display user-related Fragments; launched from {@link UserActivity}
 */

public class UserFragmentPagerAdapter extends FragmentPagerAdapter {

    //==============================================================================================
    // UserFragmentPagerAdapter fields
    //==============================================================================================

    private final int PAGE_COUNT = 7;

    private String overviewTabTitle;
    private String reposTabTitle;
    private String starredTabTitle;
    private String gistsTabTitle;
    private String pulseTabTitle;
    private String followersTabTitle;
    private String followingTabTitle;

    private Bundle userArgs;

    //==============================================================================================
    // UserFragmentPagerAdapter constructor
    //==============================================================================================

    public UserFragmentPagerAdapter(FragmentManager fragmentManager, Context context, Bundle userBundle) {
        super(fragmentManager);

        overviewTabTitle = context.getResources().getString(mb00.android.codehub.R.string.overview);
        reposTabTitle = context.getResources().getString(mb00.android.codehub.R.string.repositories);
        starredTabTitle = context.getResources().getString(mb00.android.codehub.R.string.starred);
        gistsTabTitle = context.getResources().getString(mb00.android.codehub.R.string.gists);
        pulseTabTitle = context.getResources().getString(mb00.android.codehub.R.string.pulse);
        followersTabTitle = context.getResources().getString(mb00.android.codehub.R.string.followers);
        followingTabTitle = context.getResources().getString(mb00.android.codehub.R.string.following);

        userArgs = userBundle;
    }

    //==============================================================================================
    // FragmentPagerAdapter methods
    //==============================================================================================

    @Override
    public int getCount() {
        return PAGE_COUNT;
    }

    @Override
    public Fragment getItem(int position) {
        switch(position) {
            case 0:
                UserOverviewFragment overviewFragment = new UserOverviewFragment();
                overviewFragment.setArguments(userArgs);
                return overviewFragment;
            case 1:
                UserReposFragment reposFragment = new UserReposFragment();
                reposFragment.setArguments(userArgs);
                return reposFragment;
            case 2:
                UserStarredFragment starredFragment = new UserStarredFragment();
                starredFragment.setArguments(userArgs);
                return starredFragment;
            case 3:
                UserGistsFragment gistsFragment = new UserGistsFragment();
                gistsFragment.setArguments(userArgs);
                return gistsFragment;
            case 4:
                UserPulseFragment pulseFragment = new UserPulseFragment();
                pulseFragment.setArguments(userArgs);
                return pulseFragment;
            case 5:
                UserFollowersFragment followersFragment = new UserFollowersFragment();
                followersFragment.setArguments(userArgs);
                return followersFragment;
            case 6:
                UserFollowingFragment followingFragment = new UserFollowingFragment();
                followingFragment.setArguments(userArgs);
                return followingFragment;
            default:
                return null;
        }
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch(position) {
            case 0:
                return overviewTabTitle;
            case 1:
                return reposTabTitle;
            case 2:
                return starredTabTitle;
            case 3:
                return gistsTabTitle;
            case 4:
                return pulseTabTitle;
            case 5:
                return followersTabTitle;
            case 6:
                return followingTabTitle;
            default:
                return null;
        }
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }

}

File: app/src/main/java/mb00/android/codehub/ui/user/adapter/UserFragmentPagerAdapter.kt
package mb00.android.codehub.ui.user.adapter

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.PagerAdapter
import mb00.android.codehub.R
import mb00.android.codehub.ui.user.view.*

/**
 * FragmentPagerAdapter used to display user-related Fragments; launched from [UserActivity]
 */

class UserFragmentPagerAdapter(
        private val fragmentManager: FragmentManager,
        private val context: Context,
        private val userArgs: Bundle?
) : FragmentPagerAdapter(fragmentManager) {

    private val PAGE_COUNT = 7

    private val overviewTabTitle: String = context.resources.getString(R.string.overview)
    private val reposTabTitle: String = context.resources.getString(R.string.repositories)
    private val starredTabTitle: String = context.resources.getString(R.string.starred)
    private val gistsTabTitle: String = context.resources.getString(R.string.gists)
    private val pulseTabTitle: String = context.resources.getString(R.string.pulse)
    private val followersTabTitle: String = context.resources.getString(R.string.followers)
    private val followingTabTitle: String = context.resources.getString(R.string.following)

    override fun getCount(): Int {
        return PAGE_COUNT
    }

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> getUserOverviewFragment()
            1 -> getUserReposFragment()
            2 -> getUserStarredFragment()
            3 -> getUserGistsStarred()
            4 -> getUserPulseFragment()
            5 -> getUserFollowersFragment()
            6 -> getUserFollowingFragment()
            else -> getUserOverviewFragment()
        }
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return when (position) {
            0 -> overviewTabTitle
            1 -> reposTabTitle
            2 -> starredTabTitle
            3 -> gistsTabTitle
            4 -> pulseTabTitle
            5 -> followersTabTitle
            6 -> followingTabTitle
            else -> null
        }
    }

    override fun getItemPosition(`object`: Any): Int {
        return PagerAdapter.POSITION_NONE
    }

    private fun getUserOverviewFragment(): UserOverviewFragment {
        val overviewFragment = UserOverviewFragment()
        overviewFragment.arguments = userArgs
        return overviewFragment
    }

    private fun getUserReposFragment(): UserReposFragment {
        val reposFragment = UserReposFragment()
        reposFragment.arguments = userArgs
        return reposFragment
    }

    private fun getUserStarredFragment(): UserStarredFragment {
        val starredFragment = UserStarredFragment()
        starredFragment.arguments = userArgs
        return starredFragment
    }

    private fun getUserGistsStarred(): UserGistsFragment {
        val gistsFragment = UserGistsFragment()
        gistsFragment.arguments = userArgs
        return gistsFragment
    }

    private fun getUserPulseFragment(): UserPulseFragment {
        val pulseFragment = UserPulseFragment()
        pulseFragment.arguments = userArgs
        return pulseFragment
    }

    private fun getUserFollowersFragment(): UserFollowersFragment {
        val followersFragment = UserFollowersFragment()
        followersFragment.arguments = userArgs
        return followersFragment
    }

    private fun getUserFollowingFragment(): UserFollowingFragment {
        val followingFragment = UserFollowingFragment()
        followingFragment.arguments = userArgs
        return followingFragment
    }

}
